stac_model/torch/export.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import tempfile
+from collections.abc import Sequence
+
+import torch
+import yaml
+from torch.export.dynamic_shapes import Dim
+from torch.export.pt2_archive._package import package_pt2
+
+from .utils import aoti_compile_and_extract, create_example_input_from_shape, extract_module_arg_names
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+def aoti_compile(
+    model_directory: str,
+    model_program: torch.export.ExportedProgram,
+    transforms_directory: str | None = None,
+    transforms_program: torch.export.ExportedProgram | None = None,
+) -> dict[str, list[str]]:
+    """Compiles a model and its transforms using AOTI.
+
+    Args:
+        model_directory: The directory to store the compiled model files.
+        model_program: The exported model program.
+        transforms_directory: The directory to store the compiled transforms files.
+        transforms_program: The exported transforms program.
+    """
+    model_files = aoti_compile_and_extract(
+        program=model_program,
+        output_directory=model_directory,
+    )
+    aoti_files = {"model": model_files}
+
+    if transforms_program is not None and transforms_directory is not None:
+        transforms_files = aoti_compile_and_extract(
+            program=transforms_program,
+            output_directory=transforms_directory,
+        )
+        aoti_files["transforms"] = transforms_files
+
+    return aoti_files
+
+
+@torch.no_grad()
+def export(
+    input_shape: Sequence[int],
+    model: torch.nn.Module,
+    transforms: torch.nn.Module | None = None,
```

Comment:
If multiple transforms are needed, this seems to enforce using `transforms.Compose` or some other custom `nn.Module` that combines them. Is that the intent? Is that clear enough or should more guidance be described in the parameter?

Reply:
That is the intent. @isaaccorley could you add more description on this, maybe linking to an example of transforms.Compose with torchvision?

Reply:
The only requirement is that it's an nn.Module. We have been wrapping multiple into a nn.Sequential instead of T.Compose because T.Compose isn't exportable.

Reply:
We could require that it's an nn.Sequential and not a nn.Module if that makes sense.

Reply:
I think either `nn.Module` or `nn.Sequential` should be allowed, since MLM now allows either single or array-based `pre_processing_function`. Since `nn.Sequential` is also a `nn.Module`, I think we should list them both explicitly (even if redundant), just for the clarity about single vs array of transforms. The input could also be an iterable of `nn.Module`, which the function would automatically wrap in `nn.Sequential` if that is required for the export/packaging.

Reply:
It's not required for export/packaging. Both nn.Module and nn.Sequential work fine they just have to be some class that inherits from nn.Module.

Reply:
I've added some info to the docstring on recommending to wrap multiple transforms in an nn.Sequential